pkg/clients/s3/bucketclients/lifecycleConfig.go

```diff
@@ -0,0 +1,151 @@
+	}
+
+	if len(conf.Rules) != 0 && in.config == nil {
+		return NeedsDeletion, nil
+	}
+
+	rules, err := in.generateConfiguration()
+	if err != nil {
+		return NeedsUpdate, errors.Wrap(err, "unable to create rules for bucket lifecycle reconcile")
+	}
+	if cmp.Equal(conf.Rules, rules) {
+		return Updated, nil
+	}
+	return NeedsUpdate, nil
+}
+
+func (in *LifecycleConfigurationClient) createLifecycleRule(other v1beta1.LifecycleRule) (*awss3.LifecycleRule, error) {
+	rule := awss3.LifecycleRule{
+		ID:     other.ID,
+		Status: awss3.ExpirationStatus(other.Status),
+	}
+	if other.AbortIncompleteMultipartUpload != nil {
+		rule.AbortIncompleteMultipartUpload = &awss3.AbortIncompleteMultipartUpload{
+			DaysAfterInitiation: other.AbortIncompleteMultipartUpload.DaysAfterInitiation,
+		}
+	}
+	if other.Expiration != nil {
+		date, err := time.Parse("2006-01-02T15:04:05.000Z", aws.StringValue(other.Expiration.Date))
+		if err != nil {
+			return nil, err
+		}
+		rule.Expiration = &awss3.LifecycleExpiration{
+			Date:                      &date,
+			Days:                      other.Expiration.Days,
+			ExpiredObjectDeleteMarker: other.Expiration.ExpiredObjectDeleteMarker,
+		}
+	}
+	if other.NoncurrentVersionExpiration != nil {
+		rule.NoncurrentVersionExpiration = &awss3.NoncurrentVersionExpiration{NoncurrentDays: other.NoncurrentVersionExpiration.NoncurrentDays}
+	}
+	if other.NoncurrentVersionTransitions != nil {
+		rule.NoncurrentVersionTransitions = make([]awss3.NoncurrentVersionTransition, len(other.NoncurrentVersionTransitions))
+		for tIndex, transition := range other.NoncurrentVersionTransitions {
+			rule.NoncurrentVersionTransitions[tIndex] = awss3.NoncurrentVersionTransition{
+				NoncurrentDays: transition.NoncurrentDays,
+				StorageClass:   awss3.TransitionStorageClass(transition.StorageClass),
+			}
+		}
+	}
+	if other.Transitions != nil {
+		rule.Transitions = make([]awss3.Transition, len(other.Transitions))
+		for tIndex, transition := range other.Transitions {
+			date, err := time.Parse("2006-01-02T15:04:05.000Z", aws.StringValue(transition.Date))
+			if err != nil {
+				return nil, err
+			}
+			rule.Transitions[tIndex] = awss3.Transition{
+				Date:         &date,
+				Days:         transition.Days,
+				StorageClass: awss3.TransitionStorageClass(transition.StorageClass),
+			}
+		}
+	}
+	return &rule, nil
+}
+
+func (in *LifecycleConfigurationClient) generateConfiguration() ([]awss3.LifecycleRule, error) {
+	rules := make([]awss3.LifecycleRule, len(in.config.Rules))
+	for i, v := range in.config.Rules {
+		rule, err := in.createLifecycleRule(v)
+		if err != nil {
+			return nil, err
+		}
+		rules[i] = *rule
+	}
+	return rules, nil
+}
+
+// GenerateLifecycleConfigurationInput creates the input for the LifecycleConfiguration request for the S3 Client
+func (in *LifecycleConfigurationClient) GenerateLifecycleConfigurationInput(name string) (*awss3.PutBucketLifecycleConfigurationInput, error) {
```

Comment:
```suggestion
func GenerateLifecycleConfiguration(in v1beta1.BucketParameters) []awss3.LifecycleRule {
```

With my suggestions above, I think that's how the signature for this function would look like.